zetaclient/evm/outbounds.go

```diff
@@ -0,0 +1,241 @@
+package evm
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"strings"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"github.com/zeta-chain/protocol-contracts/pkg/contracts/evm/erc20custody.sol"
+	"github.com/zeta-chain/protocol-contracts/pkg/contracts/evm/zetaconnector.non-eth.sol"
+	"github.com/zeta-chain/zetacore/pkg/chains"
+	"github.com/zeta-chain/zetacore/pkg/coin"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+	"github.com/zeta-chain/zetacore/zetaclient/compliance"
+)
+
+// PostVoteOutbound posts vote to zetacore for the confirmed outtx
+func (ob *ChainClient) PostVoteOutbound(
+	cctxIndex string,
+	receipt *ethtypes.Receipt,
+	transaction *ethtypes.Transaction,
+	receiveValue *big.Int,
+	receiveStatus chains.ReceiveStatus,
+	nonce uint64,
+	cointype coin.CoinType,
+	logger zerolog.Logger,
+) {
+	chainID := ob.chain.ChainId
+	zetaTxHash, ballot, err := ob.zetaBridge.PostVoteOutbound(
+		cctxIndex,
+		receipt.TxHash.Hex(),
+		receipt.BlockNumber.Uint64(),
+		receipt.GasUsed,
+		transaction.GasPrice(),
+		transaction.Gas(),
+		receiveValue,
+		receiveStatus,
+		ob.chain,
+		nonce,
+		cointype,
+	)
+	if err != nil {
+		logger.Error().Err(err).Msgf("PostVoteOutbound: error posting vote for chain %d nonce %d outtx %s ", chainID, nonce, receipt.TxHash)
+	} else if zetaTxHash != "" {
+		logger.Info().Msgf("PostVoteOutbound: posted vote for chain %d nonce %d outtx %s vote %s ballot %s", chainID, nonce, receipt.TxHash, zetaTxHash, ballot)
+	}
+}
+
+// IsCctxOutTxProcessed checks outtx status and returns (isIncluded, isConfirmed, error)
+// It also posts vote to zetacore if the tx is confirmed
+func (ob *ChainClient) IsCctxOutTxProcessed(cctx *crosschaintypes.CrossChainTx, logger zerolog.Logger) (bool, bool, error) {
+	// skip if outtx is not confirmed
+	nonce := cctx.GetCurrentOutTxParam().OutboundTxTssNonce
+	if !ob.isTxConfirmed(nonce) {
+		return false, false, nil
+	}
+	receipt, transaction := ob.GetTxNReceipt(nonce)
+	sendID := fmt.Sprintf("%d-%d", ob.chain.ChainId, nonce)
+	logger = logger.With().Str("sendID", sendID).Logger()
+
+	// get connector and erce20Custody contracts
+	connectorAddr, connector, err := ob.GetConnectorContract()
+	if err != nil {
+		return false, false, errors.Wrapf(err, "error getting zeta connector for chain %d", ob.chain.ChainId)
```

Comment:
Can we try to cover this line?

Reply:
https://github.com/zeta-chain/node/pull/1989/commits/9f61d5014fcbde886a962b2a93b7cb74f6177c47